EDU-12175 - Category cover: Routing (#1371)
vtexdocs/dev-portal-content: docs/guides/vtex-io/Storefront-Guides/routing.md

---
title: "Routing"
slug: "routing"
hidden: false
createdAt: "2024-09-17T14:48:17.530Z"
updatedAt: ""
excerpt: "Learn how routing is managed in VTEX IO Store Framework."
---

A route maps a URL pattern and an HTTP request method to an action. It defines how the application responds to client requests for specific endpoints.

When using [VTEX IO Store Framework](https://developers.vtex.com/docs/guides/store-framework), you do not need to manage routing and HTTP methods directly. Instead, routing and HTTP request methods are managed by either the [Rewriter](https://developers.vtex.com/docs/apps/vtex.rewriter) app or the [store builder](https://developers.vtex.com/docs/guides/vtex-io-documentation-store-builder), depending on the route type. This allows you to focus on creating and managing route paths and page templates.

> VTEX IO routes are classified into four different categories:

Reviewer: ```suggestion
Store Framework routes are classified into four different categories:
```

- **Product routes:** Related to product pages.
- **Search routes:** Related to search pages.

> - **Navigation routes:** Client-side defined routes related to VTEX IO custom paths and predefined templates, such as [department](https://github.com/vtex-apps/store/blob/master/store/routes.json#L27), [brand](https://github.com/vtex-apps/store/blob/master/store/routes.json#L21), and [category](https://github.com/vtex-apps/store/blob/master/store/routes.json#L33) routes.

Reviewer: ```suggestion
- **Navigation routes:** Client-side defined routes related to Store Framework custom paths and predefined templates, such as [department](https://github.com/vtex-apps/store/blob/master/store/routes.json#L27), [brand](https://github.com/vtex-apps/store/blob/master/store/routes.json#L21), and [category](https://github.com/vtex-apps/store/blob/master/store/routes.json#L33) routes.
```

> 
> VTEX IO routes are classified into four different categories:
> 
> - **Product routes:** Related to product pages.
> - **Search routes:** Related to search pages.
> - **Navigation routes:** Client-side defined routes related to VTEX IO custom paths and predefined templates, such as [department](https://github.com/vtex-apps/store/blob/master/store/routes.json#L27), [brand](https://github.com/vtex-apps/store/blob/master/store/routes.json#L21), and [category](https://github.com/vtex-apps/store/blob/master/store/routes.json#L33) routes.
> - **Custom routes:** Created by the user to handle custom landing pages.

Reviewer: ```suggestion

### Route management responsibilities

In Store Framework, different apps and builder manage different types of routes. Below is a table outlining the primary managers for each route type:

| Route type      | Responsibility             | Description                                                                                   |
|---------------------|------------------------------------|---------------------------------------------------------------------------------------------------|
| **Product routes**    |  Rewriter app                   | Routes related to product pages, forwarded by the Rewriter app to the rendering pipeline. |
| **Search routes**     | Rewriter app                     | Routes related to search result pages, processed and forwarded by the Rewriter app. |
| **Catalog routes**     | Rewriter app                  |Custom routes for predefined templates (e.g., [department](https://github.com/vtex-apps/store/blob/master/store/routes.json#L27), [brand](https://github.com/vtex-apps/store/blob/master/store/routes.json#L21), and [category](https://github.com/vtex-apps/store/blob/master/store/routes.json#L33)) handled by the Rewriter app and mapped to the appropriate page.|
| **Custom routes**    |  Store builder              | User-defined routes for custom landing pages or other routes that fall outside of predefined product or category routes, directly managed by the store builder and forwarded to the render server. |

```

Reviewer: De repente uma tabela assim pode ajudar a deixar mais visual e objetivo de quem é a responsabilidade de gerenciar cada tipo de rota. Se achar válido, vale verificar a necessidade de ajustes nos próximos dois parágrafos

> The Rewriter app manages the first three route types. It interprets the requested path, identifies the route type, and then forwards the route path to the rendering pipeline. Learn more in [Rewriter GraphQL API](https://developers.vtex.com/docs/apps/vtex.rewriter@1.63.0/rewriter-graphql-api).

Reviewer: ```suggestion
The [Rewriter app](https://developers.vtex.com/docs/apps/vtex.rewriter) manages the first three route types. It interprets the requested path, identifies the route type, and then forwards the route path to the rendering pipeline. Learn more in [Rewriter GraphQL API](https://developers.vtex.com/docs/apps/vtex.rewriter@1.63.0/rewriter-graphql-api).
```

> The store builder manages custom routes. Therefore, custom routes are not stored in the Rewriter app, meaning their paths are directly forwarded to the render server. To learn how to create a custom page, see the [Creating a new custom page](https://developers.vtex.com/docs/guides/vtex-io-documentation-creating-a-new-custom-page) guide.

Reviewer: ```suggestion
The [store builder](https://developers.vtex.com/docs/guides/vtex-io-documentation-store-builder) manages custom routes. Therefore, custom routes are not stored in the Rewriter app, meaning their paths are directly forwarded to the render server. To learn how to create a custom page, see the [Creating a new custom page](https://developers.vtex.com/docs/guides/vtex-io-documentation-creating-a-new-custom-page) guide.
```

### Check the builders

Check if the builders are properly installed in your Store Theme. To use builders, your account must have at least version `vtex.builder-hub@0.293.4` installed. Also, you need to specify them in the app's `manifest.json` file. Learn more in [Builders](https://developers.vtex.com/docs/guides/vtex-io-documentation-builders).

Make sure the [store builder](https://developers.vtex.com/docs/guides/vtex-io-documentation-store-builder) is configured, as it enables the development of Store Framework storefronts.

### Check Rewriter installation

To check if the [Rewriter app](https://developers.vtex.com/docs/apps/vtex.rewriter) is installed in your store, follow the guide [Listing an account's apps](https://developers.vtex.com/docs/guides/vtex-io-documentation-listing-an-accounts-apps) and search for `vtex.rewriter` in the **Installed apps** list. If it is not, install the app by following the guide [Installing an app](https://developers.vtex.com/docs/guides/vtex-io-documentation-installing-an-app).

</Steps>

> ## Guides in this section

Reviewer: Pode reaproveitar as descriptions que você criou nesta seção para atualizar o `excerpt` dos guias mencionados
